isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/services.adoc

```diff
@@ -0,0 +1,247 @@
+
+Für asynchrone Service-Aufrufe gelten dieselben Vorgaben wie für <<synchrone-service-aufrufe,synchrone Service-Aufrufe>>.
+Sie unterscheiden sich im Ablauf dahingehend, dass der Sender nicht aktiv auf die Antwort des Empfängers wartet.
+Stattdessen wird die Verarbeitung erst durch die Antwort des Empfängers wieder aufgenommen, z.B. in Form eines Callbacks.
+
+[[service-aufruf-asynchron]]
+.Ablauf eines asynchronen Service-Aufrufs
+image::software-technisch/asynchroner-service-aufruf.dn.svg[]
+
+Asynchrone Service-Aufrufe können z.B. dann eingesetzt werden, wenn eine länger dauernde Verarbeitung durch den Empfänger eine direkte Rückmeldung unmöglich macht.
+
+[[queueing]]
+=== Queueing
+
+Beim Queueing baut ein Message-Broker eine Punkt-zu-Punkt-Verbindung zwischen zwei IT-Systemen auf.
+Dies geschieht in Form einer Queue.
+Ein IT-System tritt fest als Sender auf, eines als Empfänger.
+Der Sender ist nun in der Lage, dem Empfänger über die Queue Nachrichten zu schicken.
+Die Nachrichten sind anhand eines zentral definierten Formats strukturiert.
+Der Sender enthält keine direkte Antwort vom Empfänger.
+
+[[service-aufruf-queueing]]
+.Ablauf der Kommunikation beim Queueing
+image::software-technisch/queueing.dn.svg[]
+
+Für das Queueing infrage kommende Message-Broker müssen *JMS* (Jakarta Messaging, ehemals _Java Message Service_) unterstützen.
+Queueing wird ausschließlich in der internen Servicekommunikation eingesetzt.
+
+JMS-Nachrichten bestehen aus Header, Properties und einem Body.
+Die Properties unterteilen sich noch einmal in applikationsspezifische Properties, die nur für Publisher und Subscriber Bedeutung haben, sowie provider-spezifische und Standard-Properties, die zur Verarbeitung der JMS-Nachrichten durch den Message-Broker gedacht sind.
+
+[[jms-message-aufbau]]
+.Aufbau einer JMS-Nachricht
+image::software-technisch/jms-message-aufbau.dn.svg[]
+
+Applikationsspezifische Properties enthalten Metadaten.
+Der Body enthält Nutzdaten.
+Nutzdaten werden im XML-Format übertragen und mittels XSD spezifiziert.
+
+Diese Vorgabe steht vollständig in Einklang mit der JMS-Spezifikation.
+Für die Übertragung von Nutzdaten sieht die JMS-Spezifikation fünf Formate vor.
+Die Architekturvorgabe sieht die alleinige Nutzung der Ausprägung `TextMessage` vor, die Nutzdaten als Zeichenkette erwartet.
+
+[NOTE]
+====
+Weitere Details zu JMS-Nachrichten finden sich in der JMS-Spezifikation im Kapitel https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-message-model[3. Jakarta Messaging message model].
+Besonders relevant für die Referenzarchitektur sind die Abschnitte https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-messages[3.3. Jakarta Messaging messages] sowie https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-message-body[3.11. Jakarta Messaging message body].
+====
+
+== Kommunikation mit externen Systemen
```

Comment:
ist in diesem Abschnitt sowohl die synchrone als auch die asynchrone Kommunikation mit externen Systemen definiert.
Ich lese den Abschnitt so, als hätte man hier nur die synchrone Kommunikation im Fokus.

Reply:
Es steht nirgends explizit, aber ich habe dieselbe Vermutung. Das Thema wird in einem Folgeticket behandelt.